server/supernova/sc/sc_osc_handler.cpp

```diff
@@ -3626,6 +3648,40 @@ void sc_osc_handler::do_asynchronous_command(World* world, void* replyAddr, cons
                                            shared_endpoint);
 }
 
+// called from RT thread, perform in NRT thread, free in RT thread
+template <bool realtime> void handle_message_from_RT(World* world, FifoMsg& msg) {
+    // see handle_asynchronous_command()
+    spin_lock::scoped_lock lock(system_callback_allocator_lock);
+
+    cmd_dispatcher<realtime>::fire_system_callback([=]() mutable {
+        msg.Perform();
+
+        cmd_dispatcher<realtime>::fire_rt_callback([=]() mutable { msg.Free(); });
+    });
+}
+
+void sc_osc_handler::send_message_from_RT(World* world, FifoMsg& msg) {
+    if (world->mRealTime)
+        handle_message_from_RT<true>(world, msg);
+    else
+        handle_message_from_RT<false>(world, msg);
+}
+
+// called from NRT thread, perform in RT thread, free in NRT thread
+template <bool realtime> void handle_message_to_RT(World* world, FifoMsg& msg) {
+    cmd_dispatcher<realtime>::fire_rt_callback([=]() mutable {
+        msg.Perform();
+
+        cmd_dispatcher<realtime>::fire_system_callback([=]() mutable { msg.Free(); });
+    });
+}
+
+void sc_osc_handler::send_message_to_RT(World* world, FifoMsg& msg) {
+    if (world->mRealTime)
+        handle_message_to_RT<true>(world, msg);
+    else
+        handle_message_to_RT<false>(world, msg);
+}
```

Comment:
could this be simplified to 
`handle_message_to_RT<world->mRealTime>(world, msg);`
?

Reply:
no, because `mRealTime` is a runtime variable but template arguments must be compile time constants.